### PR TITLE
[NPU] fix the grad kernel diff bug of gather op

### PR DESCRIPTION
**Pull request by xymyeah**
### PR types

Bug fixes

### PR changes

OPs

### Describe

[NPU] fix grad kernel diff bug of gather op
image

gather_grad
image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/npu/test_gather_op_npu.py [hunk 1]
```diff
@@ -14,101 +14,117 @@
...
-                (z_value == z_expected).all(),
-                True,
-                msg="z_value = {}, but expected {}".format(z_value, z_expected))
-
-    def test_backward(self):
-        # TODO(ascendrc): Test backward after add grad npu op implemented.
-        pass
+            x_np = np.array([[1, 2], [3, 4], [5, 6]]).astype('float32')
+            index_np = np.array([1, 1]).astype('int32')
+            #axis_np = np.array([1]).astype('int32')
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_gather_op_npu.py]: Please remove useless code, line 111 and line 117

**Review by liym27 — COMMENTED** [review]
Please provide the log of calling gather grad kernel

File: python/paddle/fluid/tests/unittests/npu/test_gather_op_npu.py (hunk 1 above)
Comment (xymyeah) [reply python/paddle/fluid/tests/unittests/npu/test_gather_op_npu.py]: done

**Review by liym27 — APPROVED** [review]
LGTM

**Pull request merged**